src/ripple/app/misc/impl/AccountTxPaging.cpp

```diff
@@ -0,0 +1,246 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/misc/impl/AccountTxPaging.h>
+#include <ripple/app/tx/Transaction.h>
+#include <ripple/protocol/Serializer.h>
+#include <beast/cxx14/memory.h> // <memory>
+
+namespace ripple {
+
+void
+convertBlobsToTxResult (
+    NetworkOPs::AccountTxs& to,
+    std::uint32_t ledger_index,
+    boost::optional<std::string> const& status,
+    Blob const& rawTxn,
+    Blob const& rawMeta)
+{
+    SerialIter it (rawTxn);
+    STTx::pointer txn = std::make_shared<STTx> (it);
+    std::string reason;
+
+    auto tr = std::make_shared<Transaction> (txn, Validate::NO, reason);
+
+    tr->setStatus (Transaction::sqlTransactionStatus(status));
+    tr->setLedger (ledger_index);
+
+    to.emplace_back(std::make_pair(std::move(tr),
+        std::make_shared<TransactionMetaSet> (
+            tr->getID (), tr->getLedger (), rawMeta)));
+};
+
+void
+saveLedgerAsync (std::uint32_t seq)
+{
+    Ledger::pointer ledger = getApp().getOPs().getLedgerBySeq(seq);
+    if (ledger)
+        ledger->pendSaveValidated(false, false);
+}
+
+void
+accountTxPage (
+    DatabaseCon& connection,
+    std::function<void (std::uint32_t)> const& onUnsavedLedger,
+    std::function<void (std::uint32_t,
+                        boost::optional<std::string> const&,
+                        Blob const&,
+                        Blob const&)> const& onTransaction,
+    RippleAddress const& account,
+    std::int32_t minLedger,
+    std::int32_t maxLedger,
+    bool forward,
+    Json::Value& token,
```

Comment:
The `token` parameter in particular needs to be documented. Something like: "Input/Output parameter. Input is a marker for the minLedger and seq. May be empty on input. Output is a look-ahead to the next ledgerSeq/txnSeq. May be empty on output."